x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/utils/is_path_valid.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get, isPlainObject } from 'lodash/fp';
+import type { SignalSource } from '../../../types';
+
+/**
+ * Returns true if path in SignalSource object is valid
+ * Path is valid if each field in hierarchy is object or undefined
+ * Path is not valid if ANY of field in hierarchy is not object or undefined
+ * @param path in source to check within source
+ * @param source The source document
+ * @returns boolean
+ */
+export const isPathValid = (path: string, source: SignalSource): boolean => {
+  if (!path) {
+    return false;
+  }
+  const splitPath = path.split('.');
+
+  return !splitPath.some((_, index, array) => {
+    const newPath = [...array].splice(0, index + 1).join('.');
+    const valueToCheck = get(newPath, source);
+    return valueToCheck !== undefined && !isPlainObject(valueToCheck);
```

Comment:
```suggestion
  return splitPath.every((_, index, array) => {
    const newPath = [...array].splice(0, index + 1).join('.');
    const valueToCheck = get(newPath, source);
    return valueToCheck === undefined || isPlainObject(valueToCheck);
```

Reply:
nit: Removing some of the `NOT`s reads a bit easier to me

Reply:
used `some` here as it can return earlier and do nod need to traverse the whole array if one of the properties is not valid.
But it should not be the big issue here though, if it is more readable agree to keep it